src/main/java/eu/europeana/metis/sandbox/service/util/DefaultXmlRecordProcessorService.java

```diff
@@ -0,0 +1,75 @@
+package eu.europeana.metis.sandbox.service.util;
+
+import static java.util.Objects.requireNonNull;
+
+import eu.europeana.metis.sandbox.common.exception.NonRecoverableServiceException;
+import eu.europeana.metis.sandbox.common.exception.ServiceException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.stereotype.Service;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+@Service
+class DefaultXmlRecordProcessorService implements XmlRecordProcessorService {
+
+  private final ObjectFactory<SAXParserFactory> objectFactory;
+
+  public DefaultXmlRecordProcessorService(ObjectFactory<SAXParserFactory> objectFactory) {
+    this.objectFactory = objectFactory;
+  }
+
+  @Override
+  public String getRecordId(String record) {
+    requireNonNull(record, "Record must not be null");
+
+    SAXParser parser;
```

Comment:
Why use a sax parser? Why not use xpath? It would seem simpler to me, and I think it is faster. But perhaps I'm missing something?

Reply:
No not missing anything, I have not work with XMLs for some time so when I looked around I saw DOM vs SAX but did not look at xpath, that is the explanation. I will refactor to switch to XPath, thanks